src/dotenv/main.py

```diff
@@ -305,7 +305,7 @@ def _is_interactive():
 
 
 def load_dotenv(
-    dotenv_path: Union[str, os.PathLike, None] = None,
+    dotenv_path: Union[str, 'os.PathLike[str]', 'os.PathLike[bytes]', None] = None,
```

Comment:
Is this better: `Union[str, os.PathLike[Union[str, bytes]], None]`?

Reply:
```python
dotenv_path: str | os.PathLike[str | bytes] | None = None
```

![image](https://user-images.githubusercontent.com/38415384/211128544-73240eff-3fc8-4c40-8d10-617229bebfb4.png)

is compact and nice, but currently [supported from Python 3.10 and above](https://peps.python.org/pep-0604/). So yes [typing.Optional](https://docs.python.org/3/library/typing.html#typing.Optional) would also be fine.

```python
from typing import Optional, Union
...
dotenv_path: Optional[Union[str, os.PathLike[Union[str, bytes]]] = None
```